pupil-api/src/services/pupil-authentication.service.ts

```diff
@@ -29,26 +33,53 @@ function initAzureTableService (): void {
 }
 
 export const pupilAuthenticationService = {
+
+  /**
+   * Authenticate the user against the preparedCheck table storage table
+   * @param {string} pupilPin - the pupil pin supplied by the user
+   * @param {string} schoolPin - the school password supplied by the user
+   * @param {object} tableService - optional table service driver
+   */
   authenticate: async function authenticate (pupilPin: string, schoolPin: string, tableService?: any): Promise<object> {
     // set tableService to azureTableService if not provided, but only instantiate it once.
     if (!tableService) {
       initAzureTableService()
       tableService = azureTableService
     }
 
+    // This will throw if the pupil is not found
     const { result } = await tableService.retrieveEntityAsync(authTable, schoolPin, pupilPin)
 
+    if (!path(['pinExpiresAt', '_'], result)) {
```

Comment:
is the `_` the direct value in these cases?

Reply:
Yes.  Blame Azure Table Storage.